sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/PruneFileSourcePartitionsSuite.scala

```diff
@@ -45,35 +46,27 @@ class PruneFileSourcePartitionsSuite extends PrunePartitionSuiteBase {
 
   test("PruneFileSourcePartitions should not change the output of LogicalRelation") {
     withTable("test") {
-      withTempDir { dir =>
-        sql(
-          s"""
-            |CREATE EXTERNAL TABLE test(i int)
-            |PARTITIONED BY (p int)
-            |STORED AS parquet
```

Comment:
so it's not only moving the package, but also changes some tests to not use hive tables but use data source tables instead?

Reply:
Yes, since we're moving `PruneFileSourcePartitionsSuite` out of the `hive` package, we need to remove the Hive dependency here too.

As commented in the [other thread](https://github.com/apache/spark/pull/33350#discussion_r671463471), to me it's OK to switch to use data source table here. I also digged the history of the change, and it seems at the time when this test was added (in #15569), data source table doesn't use HMS to store table metadata by default (it was added #15515 later), but instead was using `ListingFileCatalog` (?). Maybe it was for testing purpose that we created a Hive table here but then constructed a `LogicalRelation` to feed into the `PruneFileSourcePartitions` rule?

Let me know if you see concern here @cloud-fan , since you are the main author of this test and the related code :)